app/models/mixins/external_url_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ExternalUrlMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :external_urls, :as => :resource, :dependent => :destroy
+  end
+
+  def external_url=(url, user = User.current_user)
+    external_urls.where(:user => user).destroy_all
+    ExternalUrl.create!(:resource => self, :url => url, :user => user)
```

Comment:
@martinpovolny It still seems odd to me that it is configured as a `has_many` association but we are clearly treating it like a `has_one` here.  Is there reasoning behind using `has_many` that I'm not aware of?

Reply:
It's scoped by the user. There can be a different URL for different user.

One user clicking on a custom button associated with a VM can get a different link than a different user clicking the same VM>

Reply:
Right 🤦‍♂.  I kept seeing the `destroy_all` and somehow skipped the fact that it is scoped per-user. 

👍